src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/InteropHelpers.cs

```diff
@@ -316,6 +316,44 @@ internal static unsafe void FixupMethodCell(IntPtr hModule, MethodFixupCell* pCe
             }
         }
 
+#if PLATFORM_WINDOWS
+        private static unsafe IntPtr GetProcAddress(IntPtr hModule, byte* methodName, CharSet charSetMangling)
+        {
+            // First look for the unmangled name.  If it is unicode function, we are going
+            // to need to check for the 'W' API because it takes precedence over the
+            // unmangled one (on NT some APIs have unmangled ANSI exports).
+            
+            var exactMatch = Interop.mincore.GetProcAddress(hModule, methodName);
+
+            if ((charSetMangling == CharSet.Ansi && exactMatch != IntPtr.Zero) || charSetMangling == 0)
+            {
+                return exactMatch;
+            }
+
+            int nameLength = strlen(methodName);
+
+            // We need to add an extra byte for the suffix, and an extra byte for the null terminator
+            byte* probedMethodName = stackalloc byte[nameLength + 2];
+
+            for (int i = 0; i < nameLength; i++)
+            {
+                probedMethodName[i] = methodName[i];
+            }
+
+            probedMethodName[nameLength + 1] = 0;
+
+            probedMethodName[nameLength] = (charSetMangling == CharSet.Ansi) ? (byte)'A' : (byte)'W';
+
+            IntPtr probedMethod = Interop.mincore.GetProcAddress(hModule, probedMethodName);
+            if (probedMethod != IntPtr.Zero)
+            {
+                return probedMethod;
+            }
+
+            return exactMatch;
```

Comment:
A little bit behavior difference between this and CoreCLR:
in CoreCLR, if it couldn't find mangled one, it will return NULL instead of exactMatch

Reply:
I'm looking through the CoreCLR behavior and it looks like CoreCLR returns the exact match if the Unicode mangled name can't be found, and has already returned earlier if an exact match was found if it's an Ansi function. As far as I can tell, the behavior in CoreCLR is the same as here, but with different code paths (and also this code doesn't include the `__stdcall` mangling).

Reply:
You are right. 


`